### PR TITLE
CLOUDP-17884 (Mark) Add support for querying open TCP connections

### PR DESCRIPTION
**Pull request by dgolub**
@markbenvenuto 

#### QA Performed
* Used standalone test driver program to ensure that correct TCP connections are shown

### CODE REVIEW

File: netstat.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2014-2015 MongoDB, Inc.
```
Comment (markbenvenuto) [inline netstat.go]: Update year

Comment (dgolub) [inline netstat.go]: Will do throughout as a separate commit.

File: netstat.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+			return nil, err
+		}
+		entries = append(entries, entry)
+		bufPtr = (*byte)(unsafe.Pointer(uintptr(unsafe.Pointer(bufPtr)) + unsafe.Sizeof(tcpRow)))
+	}
+	return entries, nil
+}
+
+func convertIPAddress(ipAddress uint32) (string, error) {
+	buf := [16]uint16{}
```
Comment (markbenvenuto) [inline netstat.go]: You need 16 bytes, not 16 shorts.

File: wrappers/tcpmib.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+	State      uint32
+	LocalAddr  uint32
+	LocalPort  uint32
+	RemoteAddr uint32
+	RemotePort uint32
+}
+
+type MIB_TCPTABLE struct {
+	NumEntries uint32
+	//Table    []MIB_TCPROW
```
Comment (markbenvenuto) [inline wrappers/tcpmib.go]: Considering adding a comment explaining you cast raw buffers instead to walk the array.

**Comment by markbenvenuto** [comment]
Done commenting. Just know that the server also supports IPV6 even if automation does not yet.

:+1:

File: wrappers/tcpmib.go (hunk 3 above)
Comment (dgolub) [inline wrappers/tcpmib.go]: We do this in a number of places, so I'm going to leave this as is.

**Comment by dgolub** [comment]
This is solely to support displaying diagnostics when a test fails, so it's unlikely that we'll need IPv6.  We can add it later if we do.

**Comment by dgolub** [comment]
Merged 0154c13.
